src/main/java/org/zalando/nakadi/service/subscription/state/CleanupState.java

```diff
@@ -32,7 +32,11 @@ public void onEnter() {
                 getContext().unregisterSession();
 
             } finally {
-                switchState(StreamingContext.DEAD_STATE);
+                if (null != exception) {
+                    getContext().switchStateImmediately(StreamingContext.DEAD_STATE);
```

Comment:
it means that subscription will be switch to the dead state and then will be returned back here to finish clean up state. it brings behaviour when subscription is switched not from cleanup to dead state, but from inside cleanup state we go to the dead state + all the things in `taskQueue` for execution will be executed anyway.
does it make sense to clean the `taskQueue`, and add fresh task, if you need immediate switch?

Reply:
Correct, that is much more cleaner and makes the code readable. Updated in `98eb8f52`

Reply:
Correct! that is much more cleaner and makes the code better. Updated in `98eb8f52`. Thanks! :)

Reply:
👍 waiting for tests to pass